### PR TITLE
Fix Sphinx-gallery display and pin sphinx-related packages

### PR DESCRIPTION
**Pull request by Nayef211**
## Description
- Fixes the issue with the recent Sphinx-Gallery update
- Pins the versions of Sphinx-related packages
- Removed `language=None` from `docs/sources/conf.py` to resolve errors from https://github.com/sphinx-doc/sphinx/pull/10481

Inspired by changes from https://github.com/pytorch/audio/pull/2629

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Jinja2<3.1.0
 sphinx==3.5.4
 -e git+https://github.com/pytorch/pytorch_sphinx_theme.git@cece053#egg=pytorch_sphinx_theme
+sphinx==5.1.1
+sphinxcontrib.katex==0.8.6
+sphinxcontrib.bibtex==2.4.2
+sphinx_gallery==0.11.1
+nbsphinx==0.8.8
```
Comment (mthrok) [inline docs/requirements.txt:8]: No need to copy all of them from audio. AFAIK `nbsphinx`, `sphinxcontrib.katex` and `sphinxcontrib.bibtex` are not used in text repo.

File: docs/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,9 @@
 Jinja2<3.1.0
 sphinx==3.5.4
```
Comment (mthrok) [inline docs/requirements.txt:2]: dup

File: docs/requirements.txt (hunk 1 above)
Comment (Nayef211) [reply docs/requirements.txt]: Should be fixed now

**Comment by Nayef211** [comment]
Remaining test failures are for `test_download_charngram_vectors` on Windows platform. Will create a separate issue to investigate.

**Review by mthrok — APPROVED** [review]

**Pull request merged**